### PR TITLE
Capture Preferred Date (Wait time)

### PR DESCRIPTION
**Pull request by narin**
## Description
As the VA, I want to track veterans' appointment wait times when they schedule directly so that we can better understand the quality of care we are providing.

We will do this by asking the user for their preferred date and later comparing it to the actual schedule date

## Testing done
Local and unit

## Screenshots
image

## Acceptance criteria
- [ ] New page exists for 'when would you like to be seen'
- [ ] New page comes after clinic selection in direct schedule flow
- [ ] New page comes before calendar / datepicker in direct schedule flow
- [ ] New page does not show in request flow

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/vaos/containers/PreferredDatePage.jsx [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+} from '../actions/newAppointment.js';
+import { getPreferredDate } from '../utils/selectors';
+import AdditionalInfo from '@department-of-veterans-affairs/formation-react/AdditionalInfo';
+
+const initialSchema = {
+  type: 'object',
+  required: ['preferredDate'],
+  properties: {
+    preferredDate: {
+      type: 'string',
```
Comment (jbalboni) [inline src/applications/vaos/containers/PreferredDatePage.jsx]: Could also add `format: 'date'`.

File: src/applications/vaos/containers/PreferredDatePage.jsx [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+const uiSchema = {
+  preferredDate: {
+    'ui:title': "What is the earliest date you'd like to be seen?",
+    'ui:widget': 'date',
+    'ui:options': {
+      hideLabelText: true,
+    },
+    'ui:validations': [
+      (errors, preferredDate) => {
+        if (moment(preferredDate).isBefore(moment(), 'day')) {
```
Comment (jbalboni) [inline src/applications/vaos/containers/PreferredDatePage.jsx]: Weird that we don't already have a common validator for this.

**Review by jbalboni — APPROVED** [review]

File: src/applications/vaos/containers/PreferredDatePage.jsx (hunk 2 above)
Comment (narin) [reply src/applications/vaos/containers/PreferredDatePage.jsx]: Yeah.. I actually couldn't find too many examples of us using this widget, and the documentation is a bit sparse.

Comment (jbalboni) [reply src/applications/vaos/containers/PreferredDatePage.jsx]: I think we generally use the `uiSchema` in `src/platform/forms-system/src/js/definitions/date.js`, there are more instances of that.

**Review by rianfowler — APPROVED** [review]

**Pull request merged**